Marlin/Marlin_main.cpp

```diff
@@ -5100,8 +5181,8 @@ inline void gcode_M42() {
           while (angle < 0.0)     // outside of this range.   It looks like they behave correctly with
             angle += 360.0;       // numbers outside of the range, but just to be safe we clamp them.
 
-          X_current = X_probe_location - (X_PROBE_OFFSET_FROM_EXTRUDER) + cos(RADIANS(angle)) * radius;
-          Y_current = Y_probe_location - (Y_PROBE_OFFSET_FROM_EXTRUDER) + sin(RADIANS(angle)) * radius;
+          X_current = X_probe_location + cos(RADIANS(angle)) * radius;
+          Y_current = Y_probe_location + sin(RADIANS(angle)) * radius;
```

Comment:
…thus, here we're obtaining "where to put the nozzle" coordinates based on the "where to put the probe" coordinates, and these are the values tested against the movement limits.